jira/client.py

```diff
@@ -121,6 +121,18 @@ def _get_template_list(data):
     return template_list
 
 
+def _field_worker(fields=None, **fieldargs):
+    data = {}
+    if fields is not None:
+        data['fields'] = fields
+    else:
+        fields_dict = {}
+        for field in fieldargs:
+            fields_dict[field] = fieldargs[field]
+        data['fields'] = fields_dict
```

Comment:
I extracted this code from the original create_issue method, but you are right that a simpler design was available. It has been refactored in the follow up commit.